zerver/lib/narrow.py

```diff
@@ -118,10 +118,9 @@ def validate_terms(self) -> "NarrowParameter":
             "id",
             "sender",
             "group-pm-with",
-            "dm-including",
             "with",
         ]
-        operators_supporting_ids = ["pm-with", "dm"]
+        operators_supporting_ids = ["pm-with", "dm", "dm-including"]
```

Comment:
I don't think you need API changes to implement this -- the client can convert the format as needed.

(It might be a reasonable PR to extend the API to support a list of IDs for that operator, but it's be easy to have the client just generate multiple `dm-including` entries, I think?)